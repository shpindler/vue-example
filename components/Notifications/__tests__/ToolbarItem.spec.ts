import { shallowMount } from '@vue/test-utils'
import Component from '../ToolbarItem.vue'

describe('NotificationToolbarItem', () => {
  test('является экземпляром Vue.', () => {
    const wrapper = shallowMount(Component)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('отображает кол-во непросмотренных уведомлений равное count.', async () => {
    const wrapper = shallowMount(Component, {
      propsData: {
        count: 1,
      },
    })
    const countContainer = wrapper.find('[data-testid="count"]')
    expect(countContainer.text()).toContain('1')
    wrapper.setProps({ count: 2 })
    await wrapper.vm.$nextTick()
    expect(countContainer.text()).toContain('2')
  })

  test('не отображает кол-во непросмотренных уведомлений в случае count = 0.', () => {
    const wrapper = shallowMount(Component, {
      propsData: {
        count: 0,
      },
    })
    expect(wrapper.find('[data-testid="count"]').exists()).toBeFalsy()
  })

  test('по умолчанию кол-во непросмотреных уведомлений равно 0.', () => {
    const wrapper = shallowMount(Component)
    expect(wrapper.props().count).toBe(0)
  })

  describe('корректно отображается', () => {
    test('по умолчанию.', () => {
      const wrapper = shallowMount(Component)
      expect(wrapper.element).toMatchSnapshot()
    })

    test('с непросмотренными уведомлениями.', () => {
      const wrapper = shallowMount(Component, {
        propsData: {
          count: 1,
        },
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
