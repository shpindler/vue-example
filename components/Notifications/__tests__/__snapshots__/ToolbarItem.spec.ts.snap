// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationToolbarItem корректно отображается по умолчанию. 1`] = `
<workspace-toolbar-item-stub
  class="notificationsAction"
  icon="build"
  title="Уведомления"
  witharrow="true"
>
   
  <!---->
</workspace-toolbar-item-stub>
`;

exports[`NotificationToolbarItem корректно отображается с непросмотренными уведомлениями. 1`] = `
<workspace-toolbar-item-stub
  class="notificationsAction"
  icon="build"
  title="Уведомления"
  witharrow="true"
>
   
  <section
    class="notificationsAction__counter"
    data-testid="count"
  >
    
    1
  
  </section>
</workspace-toolbar-item-stub>
`;
